quadriga/src/main/webapp/WEB-INF/views/admin/addApp.html

```diff
@@ -0,0 +1,24 @@
+<html layout:decorate="~{layouts/main}">
+<head>
+<title>Quadriga 2.0</title>
+</head>
+<body>
+	<div layout:fragment="content">
+	<span th:text="${alert_msg}"></span>
+		<h2>Add new App</h2>
+
+		<form action="#" th:action="@{/admin/apps/add}" method="POST"
+			th:object="${app}">
+			<input type="hidden" name="${_csrf.parameterName}"
+				value="${_csrf.token}" /> 
```

Comment:
have you checked if you actually need this? with jsps the csrf token was automatically added. not sure if it's the same with thymeleaf